cheatsheet.js

```diff
@@ -0,0 +1,300 @@
+// (Note: Arrays start at index "0")
+console.log(myArray[0]); // Logs "10"
+console.log(myArray[2]); // Logs "false"
+console.log(myArray[3]); // Logs "bizbaz"
+console.log(emptyArray[3]); // Logs "undefined"
+
+// You can set the value of an array at an index like this:
+console.log(myArray[1]); // Logs "foobar"
+myArray[1] = 'zomg I changed!';
+console.log(myArray[1]); // Logs "zomg I changed!"
+
+// Arrays may be set at any numbered index:
+console.log(emptyArray[1]); // Logs "undefined"
+console.log(emptyArray[100]); // Logs "undefined"
+emptyArray[100] = 'now it has value';
+console.log(emptyArray[1]); // Logs "undefined"
+console.log(emptyArray[100]); // Logs "now it has value"
+
+/********************** Objects ***********************/
+
+// Objects are *named* collections of values. You can create an object like this:
+var emptyObj = {};
+var myObj = {
+  myNumber: 10,
+  myString: 'foobar',
+  myBool: true
+};
+
+// You access an object at a given "key" one of two ways:
+console.log(myObj.myNumber); // Dot Syntax – Logs "10"
+console.log(myObj['myNumber']); // Bracket Syntax – Logs "10"
+
+// Bracket syntax is useful for when an object key uses an otherwise illigal character.
+// Following the same rules as variables, object keys cannot be accessed via dot syntax
+// if he first character is not a letter, an underscore (_), or a dollar sign ($) or
+// subsequent characters are not letters, numbers, underscores, or dollar signs.
+var specialObj = {
+  '1 illegal key name!': true
+};
+console.log(specialObj['1 illegal key name!']) // true
+
+
+/*******************************************************
+ *                     Expressions                     *
+ *******************************************************/
+
+// Expressions let you do smart things with your variables!
+// They let you run logic and math equations with your data.
+
+/***************** Logic Expressions *******************/
+
+// All logic expressions will evaluate to the boolean values `true` or `false`
+// depending on the conditions to the left and right side.
+
+// The `==` operator checks for *EQUALITY* and evaluates to `true` or `false`
+console.log(true == true);     // Log: true
+console.log(true == false);    // Log: false
+console.log(10 == 20);         // Log: false
+console.log('asdf' == 'asdf'); // Log: true
+
+// The `===` operator checks for *type safe* equality and evals to `true` or `false`.
+// When using `==`, Javascript will try to convert each argument to the same type.
+console.log('10' == 10);       // Log: true
+console.log('10' === 10);      // Log: false
+console.log('asdf' == true);   // Log: true
+console.log('asdf' === true);  // Log: false
+console.log('' == false);      // Log: true
+console.log('' === false);     // Log: false
+console.log(1 == true);        // Log: true
+console.log(1 === true);       // Log: false
+console.log(0 == false);       // Log: true
+console.log(0 === false);      // Log: false
+
+// Fun Fact:
```

Comment:
I don't know if this is the right place for this explanation... but I don't know where else to put it!!

Reply:
This works here! And is a fun fact!! 😄